### PR TITLE
Reorganize scripting documentation

### PR DESCRIPTION
**Pull request by rmuir**
The current scripting documentation (https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting.html) is a bit chaotic. 

I think I made it even worse recently when elaborating on the expressions documentation, simply by adding more to the huge page.

Instead I think we should factor out language-specific stuff to its own pages. This helps to organize things better, for example all the crazy security complexities of `groovy` are not relevant to languages like `painless` and `expressions`. We still need to keep it "front and center" so that groovy users are aware, but we gotta move forward too!

I also split the table into "general purpose languages" (painless, groovy, javascript, python) and "special purpose languages" (expressions, mustache, native), along with what makes them special. 

I removed the experimental tag from `expressions`, I feel like this should be ok? I think reality is quite the opposite, its the simplest, most stable, fastest one we have. We shouldn't discourage people from using it. Its been around since lucene 4.6 and we are looking at lucene 6.0 right now :)

It still feels like we are just begging people to use groovy, since all the examples use it, but I'd rather just leave that be for now, and try to improve the "path" to the "good stuff" (expressions/painless) and improve their docs at the moment. This is just a step...

And security stuff is factored out to a dedicated page, but I made lots of links to it, including very early in the page. 

Suggestions are welcome, I barely even know how to use asciidoc. I also am still not happy with both the main page and many of the subpages, I think we can improve this a lot more.

### CODE REVIEW

**Comment by rmuir** [comment]
@jdconrad @rjernst @clintongormley if you guys have time, can you look at this change?

File: docs/reference/modules/scripting/groovy.asciidoc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            }
+        }
+    }
+}
+-----------------------------------
+
+===================================================
+
+[float]
+=== Groovy Built In Functions
```
Comment (jdconrad) [inline docs/reference/modules/scripting/groovy.asciidoc]: I'm a bit confused as to why this is provided as documentation?  Is this something we do outside of Groovy or could this also be found in the Groovy documentation?  (I realize this is reformatted so it was here before, but I was curious.)

**Comment by jdconrad** [comment]
This looks like a big improvement.  Thanks for doing this!  I'd be more comfortable if @rjernst @clintongormley also took a look.

File: docs/reference/modules/scripting/groovy.asciidoc (hunk 1 above)
Comment (rmuir) [inline docs/reference/modules/scripting/groovy.asciidoc]: Yeah, dunno either. I do know that groovy sometimes "automatically imports" things, such as when looking for a class xxx, it automatically checks `java.io.xxx` and `java.net.xxx` and so on. So maybe it "automatically static-imports" these functions from java.lang.Math?

**Comment by clintongormley** [comment]
@rmuir this is an awesome change. thanks for doing it.  I've pulled it locally to use as a starting point to do a bigger reorganisation (and tidying up some asciidoc stuff).

**Comment by rmuir** [comment]
ok, thanks in advance for any tidying up!

**Pull request closed without merging**